provider/caleta/api_client_test.go

```diff
@@ -0,0 +1,287 @@
+package caleta
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/valkyrie-fnd/valkyrie/configs"
+	"github.com/valkyrie-fnd/valkyrie/internal/testutils"
+	"github.com/valkyrie-fnd/valkyrie/rest"
+)
+
+var (
+	providerConfiguration = configs.ProviderConf{
+		URL: "http://caleta-test",
+		Auth: map[string]any{
+			"operator_id": "oid",
+			"signing_key": testingPrivateKey,
+		},
+	}
+)
+
+type mockRestClient struct {
+	rest.HTTPClientJSONInterface
+	JSONFunc func(ctx context.Context, req *rest.HTTPRequest, resp any) error
+	// PostJSONFunc func(ctx context.Context, req *rest.HTTPRequest, resp any) error
+	// GetJSONFunc  func(ctx context.Context, req *rest.HTTPRequest, resp any) error
+}
+
+// func (m mockRestClient) JSON(ctx context.Context, req *rest.HTTPRequest, resp any) error {
+// 	return m.JSONFunc(ctx, req, resp)
+// }
```

Comment:
Maybe this commented out code can be removed

Reply:
Fixed!